test/tools/jsontests/TransactionTests.cpp

```diff
@@ -159,115 +158,75 @@ json_spirit::mValue doTransactionTests(json_spirit::mValue const& _input, bool _
-}
-
-BOOST_AUTO_TEST_CASE(ttTransactionTestEip155)
-{
-	dev::test::executeTests("ttTransactionTest", "/TransactionTests/EIP155", "/TransactionTestsFiller/EIP155", dev::test::doTransactionTests);
-}
-
-BOOST_AUTO_TEST_CASE(ttTransactionTestEip155VitaliksTestsHomestead)
-{
-	dev::test::executeTests("ttTransactionTestEip155VitaliksTests", "/TransactionTests/Homestead", "/TransactionTestsFiller/Homestead", dev::test::doTransactionTests);
-}
-
-BOOST_AUTO_TEST_CASE(ttTransactionTestHomestead)
-{
-	dev::test::executeTests("ttTransactionTest", "/TransactionTests/Homestead", "/TransactionTestsFiller/Homestead", dev::test::doTransactionTests);
-}
-
-BOOST_AUTO_TEST_CASE(ttTransactionTest)
-{
-	dev::test::executeTests("ttTransactionTest", "/TransactionTests", "/TransactionTestsFiller", dev::test::doTransactionTests);
-}
-
-BOOST_AUTO_TEST_CASE(ttWrongRLPTransactionHomestead)
+class transactiontestfixture
 {
-	std::string fillersPath =  dev::test::getTestPath() + "/src/TransactionTestsFiller/Homestead";
-	if (!dev::test::Options::get().filltests)
-		dev::test::executeTests("ttWrongRLPTransaction", "/TransactionTests", "/TransactionTestsFiller/Homestead", dev::test::doTransactionTests);
-	else
+public:
+	transactiontestfixture()
 	{
-		dev::test::TestOutputHelper::initTest();
-		dev::test::copyFile(fillersPath + "/ttWrongRLPTransaction.json", dev::test::getTestPath() + "/TransactionTests/Homestead/ttWrongRLPTransaction.json");
-	}
-}
+		string casename = boost::unit_test::framework::current_test_case().p_name;
 
-BOOST_AUTO_TEST_CASE(ttWrongRLPTransaction)
-{
-	std::string fillersPath = dev::test::getTestPath() + "/src/TransactionTestsFiller";
-	if (!dev::test::Options::get().filltests)
-		dev::test::executeTests("ttWrongRLPTransaction", "/TransactionTests", "/TransactionTestsFiller", dev::test::doTransactionTests);
-	else
-	{
-		dev::test::TestOutputHelper::initTest();
-		dev::test::copyFile(fillersPath + "/ttWrongRLPTransaction.json", dev::test::getTestPath() + "/TransactionTests/ttWrongRLPTransaction.json");
+		if ((casename == "ttWrongRLPFrontier" || casename == "ttWrongRLPHomestead") && test::Options::get().filltests)
+			copyAllFilesFromFolder(casename);
+		else
+			fillAllFilesInFolder(casename);
 	}
-}
 
-BOOST_AUTO_TEST_CASE(tt10mbDataFieldHomestead)
-{
-	if (test::Options::get().all)
+	void fillAllFilesInFolder(string const& _folder)
 	{
-		auto start = chrono::steady_clock::now();
+		using path = boost::filesystem::path;
+		path fillersPath = path(test::getTestPath()) / "src/TransactionTestsFiller" / path(_folder);
+		string const filter = test::Options::get().singleTestName.empty() ? string() : test::Options::get().singleTestName + "Filler";
 
-		dev::test::executeTests("tt10mbDataField", "/TransactionTests/Homestead", "/TransactionTestsFiller/Homestead", dev::test::doTransactionTests);
+		std::vector<boost::filesystem::path> files = test::getJsonFiles(fillersPath.c_str(), filter);
```

Comment:
This can also be `const`, I guess.  If not, I'm interested in the compiler error.